packages/tenant-process/src/services/tenantService.ts

```diff
@@ -84,6 +94,34 @@ export const tenantService = {
       })
     );
   },
+
+  async updateTenantMails({
+    tenantId,
+    mailsSeed,
+    authData,
+  }: {
+    tenantId: string;
+    mailsSeed: ApiTenantMailsSeed;
+    authData: AuthData;
+  }): Promise<string> {
+    await assertResourceAllowed(tenantId, authData);
+    const tenant = await readModelService.getTenantById(tenantId);
```

Comment:
```suggestion
    const [, tenant] = await Promise.all([
      assertResourceAllowed(tenantId, authData),
      readModelService.getTenantById(tenantId),
    ]);
```

Reply:
I am not sure about this. We don't need to run those two promises concurrently. We want to execute `getTenantById` only after `assertResourceAllowed`: in case the assertion fails, we don't even need to start `getTenantById`.

Reply:
> I am not sure about this. We don't need to run those two promises concurrently. We want to execute `getTenantById` only after `assertResourceAllowed`: in case the assertion fails, we don't even need to start `getTenantById`.

You're right, let's leave it like it is then.